### ISSUE TITLE
[java] IdenticalCatchBranches false positive

### ISSUE THREAD

**Issue by mkordas** [pmd/pmd#1158]
**Affects PMD Version:** 
6.4.0

**Rule:**
`IdenticalCatchBranches`

**Description:**
`IdenticalCatchBranches` reports issue when catches are not identical

**Code Sample demonstrating the issue:**

```java
            try {
                ...
            } catch (FieldNotFound e) {
                throw new IllegalStateException("field not found", e);
            } catch (FieldException | FieldConvertError e) {
                throw new IllegalArgumentException("field exception ", e); 
            }
```

```
[IdenticalCatchBranches] 'catch' branch identical to 'FieldNotFound' branch
```

**Running PMD through:** 
Gradle

**Comment by boris-petrov** [comment]
Another false positive:

```java
...
} catch (Exception1 exception) {
	throw new Exception3("Error message 1", exception);
} catch (Exception2 exception) {
	throw new Exception3("Error message 2", exception);
}
```

**Comment by oowekyala** [comment]
Thanks guys, the fix is on the way

**Comment by oowekyala** [comment]
@krichter722 The PMD plugin uses PMD 6.4.0 by default, which doesn't contain the fix. You can use pmd 6.5.0 by specifying the maven dependencies explicitly, as described [here](https://pmd.github.io/pmd-6.5.0/pmd_userdocs_tools_maven.html#upgrading-the-pmd-version-at-runtime). Your pom changes to the following:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<groupId>de.richtercloud</groupId>
	<artifactId>pmd-identical-catchh-branches</artifactId>
	<version>1.0-SNAPSHOT</version>
	<packaging>jar</packaging>
	<properties>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<maven.compiler.source>1.8</maven.compiler.source>
		<maven.compiler.target>1.8</maven.compiler.target>
		<pmd.version>6.5.0</pmd.version> <!-- <- -->
	</properties>
	<build>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-pmd-plugin</artifactId>
				<version>3.10.0</version>
				<dependencies>
					<dependency>
						<groupId>net.sourceforge.pmd</groupId>
						<artifactId>pmd-core</artifactId>
						<version>${pmd.version}</version> <!-- <- -->
					</dependency>
					<dependency>
						<groupId>net.sourceforge.pmd</groupId>
						<artifactId>pmd-java</artifactId>
						<version>${pmd.version}</version> <!-- <- -->
					</dependency>
				</dependencies>

				<executions>
					<execution>
						<id>pmd-check</id>
						<phase>validate</phase>
						<goals>
							<goal>check</goal>
						</goals>
					</execution>
				</executions>
				<configuration>
					<printFailingErrors>true</printFailingErrors>
					<rulesets>
						<ruleset>code-analysis/ruleset.xml</ruleset>
					</rulesets>
					<analysisCache>false</analysisCache>
					<includeTests>true</includeTests>
				</configuration>
			</plugin>
		</plugins>
	</build>
</project>
```

**Comment by krichter722** [comment]
@oowekyala I see, thank you. I think I did this mistake before, sorry - afaik `maven-pmd-plugin` doesn't display the PMD version it uses by default which wouldn't hurf in my point of view. The issue is fixed in 6.5.0 as expected.

**Comment by adangel** [comment]
Sounds like an interesting feature for m-pmd-p, I've created an issue there: https://issues.apache.org/jira/browse/MPMD-269
